src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -602,11 +603,28 @@ - (NSNumber *)controllerNodeID
     return nodeID;
 }
 
+static inline void emitMetricForSetupPayload(MTRSetupPayload * payload)
+{
+    MATTER_LOG_METRIC(kMetricDarwinFrameworkDeviceVendorID, [payload.vendorID unsignedIntValue]);
+    MATTER_LOG_METRIC(kMetricDarwinFrameworkDeviceProductID, [payload.productID unsignedIntValue]);
+}
+
 - (BOOL)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
                                    newNodeID:(NSNumber *)newNodeID
                                        error:(NSError * __autoreleasing *)error
 {
+    // Track overall commissioning
+    MATTER_LOG_METRIC_BEGIN(kMetricDarwinFrameworkDeviceCommissioning);
+    emitMetricForSetupPayload(payload);
+
+    // Capture in a block variable to avoid losing granularity for metrics,
```

Comment:
This is a pretty clunky pattern... you could just submit the end metric from inside the block? There is not going to be a relevant difference in timing from submitting from the end of the block vs right after the dispatch_sync has returned.

Reply:
I am trying to capture the error code form syncRunOnWorkQueueWithBoolReturnValue and end the metric since the actual end is in https://github.com/project-chip/connectedhomeip/pull/32582/files#r1525635022

Reply:
Either way I will need to have the errorCode outside the block to capture the error here.